component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/emailotp/EmailOTPAuthenticator.java

```diff
@@ -292,19 +293,46 @@ protected void processAuthenticationResponse(HttpServletRequest request, HttpSer
         }
         String userToken = request.getParameter(EmailOTPAuthenticatorConstants.CODE);
         String contextToken = (String) context.getProperty(EmailOTPAuthenticatorConstants.OTP_TOKEN);
-        if (userToken.equals(contextToken)) {
+        long generatedTime = (long) context.getProperty(EmailOTPAuthenticatorConstants.OTP_GENERATED_TIME);
+        boolean isExpired = isExpired(generatedTime, context);
+        if (userToken.equals(contextToken) && isExpired) {
```

Comment:
If the user token equals to context token and if that token is not expired only we have to allow the user to authenticate? Is not it?